### PR TITLE
fix(AppStore): remove the unique identification of an app store by its location

### LINKED ISSUE #836: Error at the endpoints for app store handling
**Describe the bug**
I have installed a DSC version 6.2.0 and follow the documentation (https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/v6/IdsEcosystem/AppStore#step-2-create-app-store-entity) 
But when I delete this app sotre ID and try to create another one with the same arguments it is giving an error message.

**To Reproduce**
1. Create the following App Store
>POST /api/appstores
 {
  "title": "Example AppStore",
  "description": "Some description",
  "location": "https://binac.fit.fraunhofer.de/api/ids/data"
}

Response body:
image
2. Delete the App Store created
> DELETE /api/appstores/{id}
image
3. Create again the App Store
>POST /api/appstores
 {
  "title": "Example AppStore",
  "description": "Some description",
  "location": "https://binac.fit.fraunhofer.de/api/ids/data"
}
image

**Expected behavior**
It should be possible to create the App via the Swagger UI call of Step 3.

### PR DESCRIPTION
**Pull request by ghost**
First approach and proposal to remove the unique identification of an AppStore in the database based on its location information. Instead, its internal ID is now used.
- AppStore.java: removes @Column(unique = true)
- AppStoreRepository: removes findAppStoreWithLocation (AppStore-Entity cant be identified unique by its location anymore)
- AppStoreService: removes getAppStoreByLocation (not unique anymore)
- ~~**major change**: POST /api/ids/app now uses the appstoreId (URI) instead of the appstore-location, so Appstore required to be created in db before sending messages to it~~
- many small subsequent necessary adjustments

### CODE REVIEW

**Comment by ghost** [comment]
_(dont think the currently failing  e2e-test is caused by my changes)_

**Comment by juliapampus** [comment]
Thanks for the fix! I just had a look at our broker documentation. Could you please change the `ids/app` endpoint so that it expects the AppStore id - not uuid? In order to have this consistent: https://github.com/International-Data-Spaces-Association/DataspaceConnector/blob/836-appstore/docs/pages/communication/v6/ecosystem/broker.md#step-1-create-broker-entity-optional

**Comment by juliapampus** [comment]
> _(dont think the currently failing e2e-test is caused by my changes)_

No, it is not, don't worry. Those two scripts will fail until we published v7.

**Review by juliapampus — DISMISSED** [review]

File: src/main/java/io/dataspaceconnector/service/message/AppStoreCommunication.java [hunk 1]
```diff
@@ -47,20 +51,36 @@
     private final @NotNull AppAppStoreLinker appAppStoreLinker;
 
     /**
-     * Link app to app store entity.
+     * Service for the broker.
+     */
+    private final @NotNull AppStoreService appStoreService;
```
Comment (ghost) [inline src/main/java/io/dataspaceconnector/service/message/AppStoreCommunication.java]: AppStoreService : "Service for the broker."

**Review by juliapampus — APPROVED** [review]

**Review by juliapampus — DISMISSED** [review]

**Review by ronjaquensel — APPROVED** [review]

**Review by juliapampus — APPROVED** [review]

**Pull request merged**